### PR TITLE
[Improve][Jdbc] Refactor ddl change

### PR DESCRIPTION
**Pull request by hailin0**
### Purpose of this pull request

[Jdbc] Refactor ddl change

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Already exists

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/JdbcDialect.java [hunk 1]
```diff
@@ -527,281 +533,242 @@ default boolean columnExists(Connection connection, TablePath tablePath, String
...
-        String addColumnSQL =
-                buildAlterTableSql(
-                        event.getSourceDialectName(),
-                        addColumn.getSourceType(),
-                        AlterType.ADD.name(),
-                        addColumn,
-                        tableIdentifierWithQuoted,
-                        StringUtils.EMPTY,
-                        afterColumn);
+        boolean someCatalog = event.getSourceDialectName().equals(dialectName());
```
Comment (dailai) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/JdbcDialect.java]: sameCatalog or someCatalog?

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/JdbcDialect.java [hunk 2]
```diff
@@ -527,281 +533,242 @@ default boolean columnExists(Connection connection, TablePath tablePath, String
...
+            } else if (someCatalog) {
+                sqlBuilder.append(" ").append(event.getColumn().isNullable() ? "NULL" : "NOT NULL");
+            } else if (SqlType.TIMESTAMP.equals(event.getColumn().getDataType().getSqlType())) {
+                log.warn(
+                        "Default value is not supported for column {} in table {}. Skipping add column operation. event: {}",
+                        event.getColumn().getName(),
+                        tablePath.getFullName(),
+                        event);
+            } else {
+                sqlBuilder.append(" NOT NULL");
```
Comment (dailai) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/JdbcDialect.java:568]: If sink is mysql and the datatype is datetime, we set not null in alter table sql statement which may lead to '0000-00-00 00:00:00' default value that can lead to error.

Comment (hailin0) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/JdbcDialect.java:568]: Please check line 561

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/JdbcDialect.java (hunk 1 above)
Comment (hailin0) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/JdbcDialect.java]: fixed

**Review by dailai — APPROVED** [review]
LGTM

**Review by Hisoka-X — APPROVED** [review]

**Pull request merged**